src/org/omegat/core/team2/impl/GITExternalGpgSigner.java

```diff
@@ -0,0 +1,471 @@
+import org.eclipse.jgit.lib.PersonIdent;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.util.FS;
+import org.eclipse.jgit.util.StringUtils;
+import org.eclipse.jgit.util.SystemReader;
+import org.eclipse.jgit.util.TemporaryBuffer;
+
+import org.omegat.util.Log;
+import org.omegat.util.OStrings;
+import org.omegat.util.Platform;
+
+public class GITExternalGpgSigner extends GpgSigner {
+
+    // A GPG environment variable name. We remove this environment variable when calling gpg.
+	private static final String PINENTRY_USER_DATA = "PINENTRY_USER_DATA";
+
+    // For sanity checking the returned signature.
+	private static final byte[] SIGNATURE_START = "-----BEGIN PGP SIGNATURE-----"
+			.getBytes(StandardCharsets.US_ASCII);
+    private static final PathScanner FROM_PATH = new PathScanner();
+
+    // error message keys
+    private static final String ExternalGpgSigner_processInterrupted = "GPG_EXTERNAL_SIGNER_PROCESS_INTERRUPTED";
+    private static final String ExternalGpgSigner_processFailed = "GPG_EXTERNAL_SIGNER_PROCESS_FAILED";
+    private static final String ExternalGpgSigner_bufferError = "GPG_EXTERNAL_SIGNER_BUFFER_ERROR";
+    private static final String ExternalGpgSigner_environmentError = "GPG_EXTERNAL_SIGNER_ENVIRONMENT_ERROR";
+    private static final String ExternalGpgSigner_noKeyFound = "GPG_EXTERNAL_SIGNER_NO_KEY_FOUND";
+    private static final String ExternalGpgSigner_skipNotAccessiblePath =
+            "GPG_EXTERNAL_SIGNER_SKIP_NOT_ACCESSIBLE_PATH";
+    private static final String ExternalGpgSigner_cannotSearch = "GPG_EXTERNAL_SIGNER_CANNOT_SEARCH";
+    private static final String ExternalGpgSigner_signingCanceled = "GPG_EXTERNAL_SIGNER_SIGNING_CANCELED";
+    private static final String ExternalGpgSigner_noSignature = "GPG_EXTERNAL_SIGNER_NO_SIGNATURE";
+    private static final String ExternalGpgSigner_gpgNotFound = "GPG_EXTERNAL_SIGNER_GPG_NOT_FOUND";
+
+    private interface ResultHandler {
+		void accept(TemporaryBuffer b) throws IOException, CanceledException;
+    }
+
+    private static void runProcess(final ProcessBuilder process, final InputStream in,
+                                   final ResultHandler stdout, final ResultHandler stderr)
+            throws IOException, CanceledException {
+        String command = String.join(" ", process.command());
+        FS.ExecutionResult result = null;
+        int code = 0;
+        try {
+            result = FS.DETECTED.execute(process, in);
+            code = result.getRc();
+            if (code != 0) {
+                if (stderr != null) {
+                    stderr.accept(result.getStderr());
+                }
+                throw new IOException(
+                        MessageFormat.format(
+                                OStrings.getString(ExternalGpgSigner_processFailed),
+                                command, code + ": "
+                                        + toString(result.getStderr())));
```

Comment:
There is an override of `OStrings.getString` to which you can pass additional arguments that will be put into the string via `MessageFormat.format`.

If you prefer to keep this style (using `MessageFormat.format` on the "outside") because it's closer to the original EGit code, then that's OK.